flocker/node/_docker.py

```diff
@@ -534,6 +534,8 @@ def _list():
                     # have been removed in another thread.
                     if e.response.status_code == NOT_FOUND:
                         continue
+                    else:
+                        raise
```

Comment:
It's untested, yes; mostly just a bugfix I noticed in passing. In order to test it though I'd need to... trigger an error in Docker? seems difficult to do in practice. Or monkeypatch I guess. If you think it's worth doing that I can add a est.